snapcraft.yaml

```
@@ -10,7 +10,41 @@ build-base: core22
 grade: devel
 
 parts:
+  probert-deb:
+    plugin: nil
+    source: https://github.com/canonical/probert.git
+    source-type: git
+    source-branch: server/jammy
+    override-pull: |
+      snapcraftctl pull
+      # install build dependencies
+      export DEBIAN_FRONTEND=noninteractive
+      export DEBCONF_NONINTERACTIVE_SEEN=true
+      sudo -E apt-get build-dep -y ./
+    override-build: |
+      # unset the LD_FLAGS and LD_LIBRARY_PATH vars that snapcraft sets for us
+      # as those will point to the $SNAPCRAFT_STAGE which on re-builds will
+      # contain things like libc and friends that confuse the debian package
+      # build system
+      # TODO: should we unset $PATH to not include $SNAPCRAFT_STAGE too?
+      unset LD_FLAGS
+      unset LD_LIBRARY_PATH
+      # run the real build (but just build the binary package, and don't
+      # bother compressing it too much)
+      dpkg-buildpackage -b -uc -us -Zgzip -zfast
+      cp ../probert-{network,common}*.deb $SNAPCRAFT_PART_INSTALL
+    stage:
+      - -probert*.deb
+    override-prime: |
```

I think it is appropriate to add the same comment as in the consoleconf-deb part:
```suggestion
    # XXX: Dirty hacks to enable building core22 on non-focal systems.
    # See below for details.
    override-prime: |
```